writing-your-first-es2015-javascript-module.md

```diff
@@ -0,0 +1,140 @@
+This brought my attention to a problem similar to the one that nyc was originally built to solve ...
+
+There are many different approaches folks use to wire up ES2015 to ES5 transpilation, the methods use a variety
+of different build tools (_gulp_, _grunt_, _webpack_, _browserify_), and the paradox of choice can be overwhelming.
+
+In the process of adding Babel support to nyc I took notes, my goal being to outline a
+_dead simple_ approach for writing ES2015 modules that work on older versions of Node.js: using
+npm scripts instead of build tools, keeping library dependencies to a minimum, and avoiding
+build steps where possible.
+
+ES2015 introduces many [awesome language features](https://babeljs.io/docs/learn-es2015/) to your JavaScript repertoire,
+and I hope this tutorial will convince you to dive in and start playing.
+
+For the purpose of this discussion I've created the ES2015 module yarsay, clone it and follow along:
+
+[https://github.com/bcoe/yarsay](https://github.com/bcoe/yarsay)
+
+## Installing and Configuring Babel
+
+To get up and running you'll first want to install the babel development dependencies:
+
+```sh
+npm i babel-cli babel-core babel-preset-es2015 --save-dev
+```
+
+Let's look at each of these:
+
+* *babel-core:* the bare-bones babel compiler (without the plugins that make it actually transpile).
+* *babel-preset-es2015:* provides the set of plugins necessary to transpile ES2015 code.
+* *babel-cli:* the command-line tool used to package your module for publication to npm.
+
+> transpilation refers to converting our ES2015 source code to ES5, a flavor of JavaScript that works on older Node.js platforms.
+
+Once you have the babel dependencies installed, the next step is to create `.babelrc` file in the root of
+your project to configure things:
+
+```json
+{
+  "presets": ["es2015"]
+}
+```
+
+This configuration tells Babel that we want to use the [babel-preset-es2015](http://babeljs.io/docs/plugins/preset-es2015/) module that we installed.
+
+Having installed the Babel dependencies, let's next look at the directory structure of our `yarsay` module:
+
+## Structuring Your Project
+
+For my [ES2015 sample project](https://github.com/bcoe/yarsay) I used the following directory structure:
+
+* `src/`: contains the pre-transpiled ES2015 source files.
+* `lib/`: this directory contains the transpiled ES5 code (you probably shouldn't
+  add this directory to source control).
+* `test/`: the test directory (the tests exercise the ES2015 code in the
+  `/src` directory rather than the transpiled code).
+
+## Unit Tests and Coverage
+
+Babel provides the module `babel-register` which hooks into Node's require statement. This hook compiles your ES2015 code on the fly as your project needs it. `babel-register` is great for writing unit-tests, allowing you to exercise your ES2015 code directly without executing your build step.
```

Comment:
[nit] Perhaps wrap the word "require" in formatting, to indicate code.

[nit] I would use "unit tests" (without the dash) instead of "unit-tests".